API.md

```diff
@@ -0,0 +1,95 @@
+API Calls
+=========
+
+Add a File
+----------
+/siafile/ POST
+
+Send JSON:
+```
+{
+  title: STRING,
+  filename: STRING,
+  description: STRING,
+  content: STRING,
+  tags: [STRING],
+}
+```
+
+Returns:
+```
+{
+  _id: bson.ObjectId,
+  title: STRING,
+  filename: STRING,
+  description: STRING,
+  content: STRING,
+  tags: [STRING],
+  uploadedTime: TIME,
+}
+```
+
+Remove A File
+-------------
+/siafile/ DELETE
+Send JSON:
+Send a hash of the Sia file and credentials using basic auth. Note only Admins
+will have delete priviliges at first.
+And the hash is of the Sia file.
+
+```
+{
+  id: STRING,
+}
+```
+
+Returns:
+
+```
+{
+  success: BOOL
+}
+```
+
+Querying for a file
+-------------------
+
+Search
+
+/siafile/search/ GET
+
+Parameters query string to search by
+
+
+Returns:
+
+```
+[
+{
+  _id: bson.ObjectId,
+  title: STRING,
+  filename: STRING,
+  description: STRING,
+  tags: [STRING],
+  uploadedTime: TIME,
+}
+]
+```
+
+Get Sia File
+------------
+/siafile/<ID> GET
+
+Returns:
+
+```
+{
+  _id: bson.ObjectId,
+  title: STRING,
+  filename: STRING,
+  description: STRING,
+  content: STRING,
+  tags: [STRING],
+  uploadedTime: TIME,
+}
+```
```

Comment:
We may want to make the names of all the fields start with a capital letter, in case the go program gets more complex and starts using multiple packages. It's unconventional, but it makes compatibility with golang a lot easier.

Also, I'm not sure about '_id' as a field name, the '_' might cause problems down the line.